client/src/components/Dialog/PostDialog/PostDialog.js

```diff
@@ -56,41 +79,66 @@ class PostDialog extends React.Component {
 
     // File pond , handle image upload
     onUploadImages = fileItems => {
-        this.setState({
-            files: fileItems.filter(fileItem => {
-                if (fileItem.file.type.toString() === 'image/jpeg') {
-                    return fileItem.file;
-                }
-                return false;
-            })
-        });
-        console.log(this.state.files);
+        let filterFiles = fileItems.filter(
+            fileItem => fileItem.file.type.toString() === 'image/jpeg'
+        );
+        this.setState({ files: filterFiles.map(filterFile => filterFile.file) });
     };
 
     // Create post
-    onCreatePress = () => {
+    onCreatePress = async () => {
+        const username = this.props.location.pathname.split('/')[2];
+        const { title, tags, files = '', description, destination } = this.state;
         if (this.state.title.length < 3) {
             this.setState({ titleError: 'Title must be greater than 3 characters' });
         } else {
-            this.setState({ titleError: '' });
-            this.props.history.push('/');
+            try {
+                const formData = new FormData();
+                formData.append('title', title);
```

Comment:
i think you should be able to add board id here, if thats what your were referring to in the PR description.
formdata.append('board', someboard._id)